lib/ansible/modules/command.py

```diff
@@ -258,8 +216,6 @@ def main():
             executable=dict(),
             creates=dict(type='path'),
             removes=dict(type='path'),
-            # The default for this really comes from the action plugin
-            warn=dict(type='bool', default=True),
```

Comment:
Outright removing this will causes tasks using `warn: no` to fail. We probably need to keep it in the argument spec, just don't handle it anywhere to avoid breaking things for people.

Reply:
@samdoran Should we show a deprecation warning if it's passed, so we don't have to keep it around forever?

Reply:
Adding a deprecation warning is the only safe path to removing this. Otherwise, it will need to stick around forever to avoid breaking playbooks.

Reply:
This would be a simple way to deal with it, though the warning message text would a bit confusing.
```python
warn=dict(type='bool', default=True, removed_in_version='2.12'),
```

Reply:
@samdoran I added the deprecation warning after posting that comment, not sure if you saw. I did it using `module.deprecate()` instead so that the error message made more sense.